workspaces/telepath-js/source/socket-io-channel.js

```diff
@@ -0,0 +1,76 @@
+import base64url from 'base64url'
+import timeoutCallback from 'timeout-callback'
+
+export class SocketIOChannel {
+  constructor (socketFactoryMethod) {
+    this.socketFactoryMethod = socketFactoryMethod
+    this.pendingNotifications = []
+    this.setupDone = false
+  }
+
+  async start (channelID, onNotification, onError, timeout = 30000) {
```

Comment:
It is better to use named parameters ({ ... }) here, especially since there are four.